API Blueprint Specification.md

```diff
@@ -478,6 +487,149 @@ Name and description of the API
 
 ---
 
+<a name="def-authentication-section"></a>
+## Authentication section
+- **Parent sections:** none, [Action section](#def-action-section)
+- **Nested sections:** [`1` Authentication schemes section](#def-authenticationschemes-section), [`1` Response section](#def-response-section)
+- **Markdown entity:** header, list
+- **Inherits from**: none
+
+#### Definition
+Defined by the `Authentication` keyword in Markdown header entity.
+
+    # Authentication
+
+**-- or --**
+
+Defined by the `Authentication` keyword in Markdown list entity followed by optional comma seperated list of scheme names which is prefixed by a colon.
+
+    + Authentication: <list>
+
+> **NOTE:** The second notation is only used when the parent section is an [Action section](#def-action-section).
+
+#### Description
+Based on the parent section, the usage of this section differs as shown below.
+
+#### API Authentication section
+Description of the authentication support for the whole API.
+
+This section **should** include one [Authentication schemes section](#def-authenticationschemes-section) and one [Response section](#def-response-section).
+
+The [Authentication schemes section](#def-authenticationschemes-section) defines all the types of authentication schemes supported by the API. Where as, the [Response section](#def-response-section) describes the error response given by an authentication protected endpoint when no authentication is provided with the request.
+
+##### Example
+
+    ## Authentication
+    This API supports different kinds of the authentication.
+
+    + Authentication Schemes
+        + ...
+
+    + Response 403 (application/json)
+
+            {
+              "message": "Authentication required",
+              "type": "error"
+            }
+
+#### Action Authentication section
+Description of the authentication support for a particular action in the API.
+
+If defined, the respecitve action of the [Action section](#def-action-section) in the API can only be accessed using a vaild authentication scheme. If a list of scheme names is provided after the `Authentication` keyword, the respective action in the API does not allow other authentication schemes.
```

Comment:
A solution to this would be to emulate how `Attributes` is done currently. Allow `Authentication` section to be in either `Action` section or a `Request` section. If it is in `Action` section, it will be default for all the child requests while allowing an `Authentication` section in the request to override it.

We don't need to have a new keyword `Authenticated` or add `optional` suffixes. What do you say?